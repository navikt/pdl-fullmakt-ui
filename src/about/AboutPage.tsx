import * as React from 'react';
import './AboutPage.css';
import NorwegianTranslation from './AboutPage_no.html';
import EnglishTranslation from './AboutPage_en.html';

export const AboutPage = (props: any) => {
  const Translated =
    (props.language ? props.language : localStorage.getItem('language') || 'no') === 'no'
      ? NorwegianTranslation
      : EnglishTranslation;

  return (
    <div>
      <main>
        <div>
          <div className="row-centered">
            <div className="aboutPanel">
              <div>
                <Translated {...props} />
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

export default AboutPage;
