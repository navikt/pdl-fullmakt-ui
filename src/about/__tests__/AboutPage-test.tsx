import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import { AboutPage } from '../AboutPage';
import AboutPageEn from '../AboutPage_en.html';
import AboutPageNo from '../AboutPage_no.html';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('AboutPage', () => {
  it('should render norwegian text', () => {
    const wrapper = shallow(<AboutPage />);
    expect(wrapper.find(AboutPageNo).length).toBe(1);
    expect(wrapper.find(AboutPageEn).length).toBe(0);
  });

  it('should render english text', () => {
    const wrapper = shallow(<AboutPage language={'en'} />);
    expect(wrapper.find(AboutPageEn).length).toBe(1);
    expect(wrapper.find(AboutPageNo).length).toBe(0);
  });

  it('should have norwegian text', () => {
    const wrapper = shallow(<AboutPageNo />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should have english text', () => {
    const wrapper = shallow(<AboutPageEn />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
