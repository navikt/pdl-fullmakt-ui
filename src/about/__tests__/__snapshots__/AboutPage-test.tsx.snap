// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage should have english text 1`] = `
<div>
  <h2>
    Data Catalog
  </h2>
  <br />
  <br />
  <p>
    Data Catalog.
  </p>
  <br />
  <br />
</div>
`;

exports[`AboutPage should have norwegian text 1`] = `
<div>
  <h2>
    Datakatalog
  </h2>
  <br />
  <br />
  <p>
    Datakatalog.
  </p>
  <br />
  <br />
</div>
`;
